script/deploy/mainnet/M2_Mainnet_Upgrade.s.sol

```diff
@@ -0,0 +1,308 @@
+    uint256 timelockEta = 1712214000;
+
+    function test_queueUpgrade() external {
+        _parseDeployedContracts("script/output/mainnet/M1_deployment_mainnet_2023_6_9.json");
+        _parseInitialDeploymentParams("script/configs/mainnet/M2_mainnet_upgrade.config.json");
+
+        // TODO: fill in correct addresses
+        // simulate deploying contracts
+        _deployImplementationContracts();
+        // confirmed correct address
+        address beaconChainOracle = 0x343907185b71aDF0eBa9567538314396aa985442;
+
+        Tx[] memory txs = new Tx[](11);
+        txs[0] = Tx(
+            address(eigenLayerProxyAdmin),
+            0,
+            abi.encodeWithSelector(
+                ProxyAdmin.upgrade.selector, 
+                TransparentUpgradeableProxy(payable(address(delegationManager))), 
+                delegationManagerImplementation
+            )
+        );
+
+        txs[1] = Tx(
+            address(eigenLayerProxyAdmin),
+            0,
+            abi.encodeWithSelector(
+                ProxyAdmin.upgrade.selector, 
+                TransparentUpgradeableProxy(payable(address(slasher))), 
+                slasherImplementation
+            )
+        );
+
+        txs[2] = Tx(
+            address(eigenLayerProxyAdmin),
+            0,
+            abi.encodeWithSelector(
+                ProxyAdmin.upgrade.selector, 
+                TransparentUpgradeableProxy(payable(address(strategyManager))), 
+                strategyManagerImplementation
+            )
+        );
+
+        txs[3] = Tx(
+            address(eigenLayerProxyAdmin),
+            0,
+            abi.encodeWithSelector(
+                ProxyAdmin.upgrade.selector, 
+                TransparentUpgradeableProxy(payable(address(delayedWithdrawalRouter))), 
+                delayedWithdrawalRouterImplementation
+            )
+        );
+
+        txs[4] = Tx(
+            address(eigenLayerProxyAdmin),
+            0,
+            abi.encodeWithSelector(
+                ProxyAdmin.upgrade.selector, 
+                TransparentUpgradeableProxy(payable(address(eigenPodManager))), 
+                eigenPodManagerImplementation
+            )
+        );
+
+        txs[5] = Tx(
+            address(eigenPodBeacon),
+            0,
+            abi.encodeWithSelector(
+                UpgradeableBeacon.upgradeTo.selector, 
+                eigenPodImplementation
+            )
+        );
+
+        // unpause everything on DelegationManager
+        txs[6] = Tx(
+            address(delegationManager), 
+            0, // value
+            abi.encodeWithSelector(Pausable.unpause.selector, 0)
+        );
+
+        // set the min withdrawal delay blocks on the DelegationManager
+        txs[7] = Tx(
+            address(delegationManager), 
+            0, // value
+            abi.encodeWithSelector(DelegationManager.setMinWithdrawalDelayBlocks.selector, DELEGATION_MANAGER_MIN_WITHDRAWAL_DELAY_BLOCKS)
+        );
```

Comment:
I assume these transactions are executed en masse, but in case they aren't (or just to avoid footgun) - let's swap the order of these two transactions and make sure the withdrawal delay is configured before we unpause the contract.

Reply:
they are all completed as part of a single transaction. I can change the order though

Reply:
adjusted the ordering in https://github.com/Layr-Labs/eigenlayer-contracts/pull/480/commits/9b7a0c33c94c20f2e5bb8abfacf115e15bf24426